internal/credentials/kubernetes/acr/workload_identity.go

```diff
@@ -0,0 +1,330 @@
+package acr
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+	"github.com/patrickmn/go-cache"
+	v1 "k8s.io/api/authentication/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/credentials"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+const (
+	accessTokenUsername  = "00000000-0000-0000-0000-000000000000"
```

Comment:
Is the username allowed to be anything?

Reply:
No, from what I can see in the [docs](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-authentication?tabs=azure-cli#az-acr-login-with---expose-token) the user name needs to be exactly that string when using short-lived tokens.